libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1976,18 +1976,30 @@ IRVariable IRGeneratorForStatements::readFromLValue(IRLValue const& _lvalue)
 					")\n";
 		},
 		[&](IRLValue::Memory const& _memory) {
-			if (_memory.byteArrayElement)
-				define(result) <<
-					m_utils.cleanupFunction(_lvalue.type) <<
-					"(mload(" <<
-					_memory.address <<
-					"))\n";
+			if (auto const* functionType = dynamic_cast<FunctionType const*>(&_lvalue.type))
```

Comment:
Should we create a util function for that? There is already something very similar in YulUtilFunctions

Reply:
I'm just looking at the usage - I think I can just make ``readMemory`` always do the cleanup (resp. change ``readFromMemoryOrCalldata`` to cleaning up from memory and validating from calldata) - I think this here is actually the only usage of it. I'll verify.

Reply:
Changed that now instead.